endpoints/openrtb2/auction.go

```diff
@@ -1381,6 +1381,8 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 	// assume that the request.imp data did not change when applying the Stored BidRequest.
 	for i := 0; i < len(impIds); i++ {
 		resolvedImp, err := jsonpatch.MergePatch(storedImps[impIds[i]], imps[idIndices[i]])
+		//writing stored imp data back to imp.ext
+		resolvedImp, err = jsonparser.Set(resolvedImp, storedImps[impIds[i]], "ext", "prebid", exchange.StoredRequestAttributes)
```

Comment:
I'm afraid you can't use `jsonparser.Set`. That's not a production ready API and we have encountered problems with it before.  `jsonparser` should be used for reads only.

Reply:
Rafactored according to discussion.